test/event_stream/subscriber_dsl_test.rb

```diff
@@ -0,0 +1,54 @@
+require_relative '../test_helper'
+
+module EventStream
+  class SubscriberDSLTest < Minitest::Should::TestCase
+
+    class MySubscriber
+      include SubscriberDSL
+
+      class << self
+        attr_accessor :events
+      end
+
+      on(:event) { |event| self.events << event  }
+
+      def a_method(event)
+        self.class.events << event
+      end
+    end
+
+    context 'The Subscriber DSL' do
+      setup do
+        MySubscriber.events = []
+      end
+
+      context 'the default stream' do
+        setup do
+          EventStream.clear_subscribers
```

Comment:
> The fact we have to rely on global state here seems less than ideal

It's only 'kind_of' global, because it's the default stream. If you want to use another stream you can, as per the test below.

> subscribe_to
This could be cool to have, but I think it's nice to have the subscribers able to declare which stream they are using

> can we subscribe to multiple event streams?
not using this DSL, but it would easily work, e.g. if `subscribe_to` existed or without using the DSL

> Then I'm not sure what the clean up would look like in order to prevent a memory leak

Everything accumulates in the stream so it should be fine.